Source/missiles.cpp

```diff
@@ -565,11 +565,11 @@ BOOL MonsterTrapHit(int m, int mindam, int maxdam, int dist, int t, BOOLEAN shif
 			if (resist) {
 				PlayEffect(m, 1);
 			} else if (monster[m]._mmode == MM_STONE) {
-				if (m > 3)
+				if (m >= MAX_PLRS)

```

Comment:
Bin exact if changed to:
```suggestion
				if (m > MAX_PLRS - 1)
```
Same for the next one.